python-frontend/src/main/java/org/sonar/python/semantic/v2/types/ProgramStateTypeInferenceVisitor.java

```diff
@@ -60,9 +61,9 @@ public void visitFunctionDef(FunctionDef pyFunctionDefTree) {
   public void visitQualifiedExpression(QualifiedExpression qualifiedExpression) {
     scan(qualifiedExpression.qualifier());
     if (qualifiedExpression.name() instanceof NameImpl name) {
-      qualifiedExpression.qualifier()
```

Comment:
If I understand correctly, this used to crash with an NPE if the qualifier type is `null`.

Do you have a reproducer when this can happen?
I think it would be good to have such case documented and tested (at least to see whether it is an edge case or something currently unsupported that we eventually need to look at).